core/src/main/java/com/microsoft/applicationinsights/internal/config/AddTypeXmlElement.java

```diff
@@ -32,7 +32,7 @@
 
     private String type;
 
-    private ArrayList<ParamXmlElement> paramElements = new ArrayList<>();
+    private final ArrayList<ParamXmlElement> paramElements = new ArrayList<>();
```

Comment:
not sure about this.  are you sure this will be an empty list?

Reply:
`final` means you can't re-assign it, but doesn't mean you can't mutate the ArrayList